### PR TITLE
Add WordCount component

### PR DESCRIPTION
**Pull request by ourmaninamsterdam**
Adds (approximate) word count functionality to the Layout view on the Article page.

screen shot 2015-02-06 at 11 45 12

### CODE REVIEW

File: app/assets/javascripts/comfortable_mexican_sofa/admin/modules/WordCount.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+  WordCountProto._cacheComponentElements = function() {
+    this.config.selectors.display = '[data-dough-wordcount-display="'+ this.$el.attr('data-dough-wordcount-target') +'"]';
+    this.$display = $('body').find(this.config.selectors.display);
+  };
+
+  WordCountProto.updateDisplay = function(input, mode) {
+    mode = mode || 'html';
+    this.$display[mode === 'markdown'? 'addClass' : 'removeClass'](this.config.selectors.hiddenClass)
+      .text(this.displayText.replace('{count}', this.countWords(input)));
```
Comment (tomas-stefano) [inline app/assets/javascripts/comfortable_mexican_sofa/admin/modules/WordCount.js]: Can we extract this to more variables (the mode statement and the display text replacement). Imho, will became more easier to read.

File: spec/javascripts/tests/WordCount_spec.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+                  '$~callout' +
+                  'Did you know?' +
+                  '~$' +
+
+                  '^ Cash savings comparison tables^';
+
+      this.component.init();
+
+      this.component.$el.text(text).trigger('input');
+      expect(this.component.$display.eq(0)).to.have.text('You have approximately 70 words!');
```
Comment (tomas-stefano) [inline spec/javascripts/tests/WordCount_spec.js]: Can `this.component.$display.eq(0)` be a variable?

File: spec/javascripts/tests/WordCount_spec.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    beforeEach(function(done) {
+      this.component = new this.WordCount(this.$fixture);
+      done();
+    });
+
+    it('should display the approximate number of words entered', function() {
+      this.component.init();
+
+      this.component.$el.text('foo bar baz').trigger('input');
+      expect(this.component.$display.eq(0)).to.have.text('You have approximately 3 words!');
```
Comment (tomas-stefano) [inline spec/javascripts/tests/WordCount_spec.js]: Can `this.component.$display.eq(0)` be a variable too?

File: spec/javascripts/tests/WordCount_spec.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
+describe.only('WordCount', function(eventsWithPromises) {
```
Comment (tomas-stefano) [inline spec/javascripts/tests/WordCount_spec.js]: Remove the only function. I need to add the only keyword on the pre-commit for javascript files.

**Comment by tomas-stefano** [comment]
Minor comments. The rest is great. Great work mate :beer:

**Comment by tomas-stefano** [comment]
LGTM now :+1:

**Comment by dougdroper** [comment]
:+1:

**Pull request merged**